libraries/AP_HAL_Linux/Util.cpp

```diff
@@ -305,3 +305,41 @@ bool Util::get_random_vals(uint8_t* data, size_t size)
     close(dev_random);
     return true;
 }
+
+bool Util::parse_cpu_set(const char *str, cpu_set_t *cpu_set)  const
+{
+    unsigned long cpu1, cpu2;
+    char *endptr, sep;
+
+    CPU_ZERO(cpu_set);
+
+    do {
+        cpu1 = strtoul(str, &endptr, 10);
+        if (str == endptr) {
+            return false;
+        }
+
+        str = endptr + 1;
+        sep = *endptr;
+        if (sep == ',' || sep == '\0') {
+            CPU_SET(cpu1, cpu_set);
+            continue;
+        }
+
+        if ((sep != '-') && (sep != ':')) {
```

Comment:
: is wrong here... it doesn't have the same meaning in isolcpus so it will just create confusion. ":" is not much used and it's meaning is to add a param after that with the 2/25 syntax, meaning "use first 2 cpus every 25 cpus". On a NUMA system this is useful for example to reserve the first cpu of every cpu socket. Considering 4 sockets with 32 cpus each you can do 0-127:1/32.

Not very useful in this context.

Reply:
understood ... done